src/main/java/com/icthh/xm/ms/timeline/service/cassandra/TimelineServiceCassandraImpl.java

```diff
@@ -0,0 +1,122 @@
+package com.icthh.xm.ms.timeline.service.cassandra;
+
+import com.icthh.xm.commons.tenant.TenantContextHolder;
+import com.icthh.xm.commons.tenant.TenantContextUtils;
+import com.icthh.xm.ms.timeline.domain.XmTimeline;
+import com.icthh.xm.ms.timeline.domain.ext.IdOrKey;
+import com.icthh.xm.ms.timeline.repository.cassandra.EntityMappingRepository;
+import com.icthh.xm.ms.timeline.repository.cassandra.TimelineCassandraRepository;
+import com.icthh.xm.ms.timeline.service.TimelineService;
+import com.icthh.xm.ms.timeline.web.rest.vm.TimelinePageVM;
+import org.apache.commons.lang3.StringUtils;
+
+import java.time.Instant;
+
+public class TimelineServiceCassandraImpl implements TimelineService {
+    private TimelineCassandraRepository timelineRepository;
+    private EntityMappingRepository entityMappingRepository;
+    private TenantContextHolder tenantContextHolder;
+
+    public TimelineServiceCassandraImpl(
+                    TimelineCassandraRepository timelineRepository,
+                    EntityMappingRepository entityMappingRepository,
+                    TenantContextHolder tenantContextHolder) {
+        this.timelineRepository = timelineRepository;
+        this.entityMappingRepository = entityMappingRepository;
+        this.tenantContextHolder = tenantContextHolder;
+    }
+
+    /**
+     * Get page with timelines.
+     *
+     * @param userKey   the user key
+     * @param idOrKey   the entity id or key
+     * @param dateFrom  the date from
+     * @param dateTo    the date to
+     * @param operation the operation
+     * @param next      the next page code
+     * @param limit     the limit per page
+     * @return page with timelines and next page code
+     */
+    @Override
+    public TimelinePageVM getTimelines(String msName,
+                                       String userKey,
+                                       String idOrKey,
+                                       Instant dateFrom,
+                                       Instant dateTo,
+                                       String operation,
+                                       String next,
+                                       int limit) {
+
+        if (idOrKey != null) {
+            IdOrKey idOrKeyObj = IdOrKey.of(idOrKey);
+
+            Long id;
```

Comment:
Why not use something like this:
```
Long id = idOrKeyObj.isId() ? idOrKeyObj.getId() : entityMappingRepository.getIdByKey(idOrKeyObj.getKey(),
                                TenantContextUtils.getRequiredTenantKeyValue(tenantContextHolder));
```

Reply:
it was old code from previous task
but fixed anyway